### PR TITLE
Benchmark: Enable 9.6, 9.8

### PR DESCRIPTION
**Pull request by soulomoon**
~~  Enable 4 version to do bench seems to be taking too long, considering picking two: highest and lowest  ~~

Fix bench for newer ghc versions
The following have been done:
1. No longer use the implicit-hie to generate the hie.yaml for the bench examples and in favor of using "cradle:\n cabal:\n", seems to be working with modern cabal.
2. upgrade benchmark to use 9.6, 9.8 (The latest two we support for now).
3. upgrade bench examples to `Cabal version: 3.10.2.1, lsp-types version: 2.1.1.0`
4. fix minor error that `*.hp` files duplicates its extension name

### CODE REVIEW

**Comment by michaelpj** [comment]
"Two highest" would also be reasonable, as those are the ones we are going to support for longest. But "highest and lowest" captures the most difference, so is maybe indeed the most interesting. If we do this we should write it down.

**Comment by soulomoon** [comment]
Some thing problematic with the hie.yaml generated by implicit-hie for lsp-types-2.1.1.0.
Instead of generate the hie.yaml with implicite-hie, fill hie.yaml simply with `"cradle:\n cabal:\n"` seems
to be work fow more examples now adays. And so I did.

Bench for 4 version: Total duration
[51m 28s](https://github.com/haskell/haskell-language-server/actions/runs/8178350258/usage).
Acceptable?

**Review by jhrcek — COMMENTED** [review]
Great work!
I'm not really sure which option is the best (2 highest vs. newest+oldest) but having all 4 seems too much.

From maintenance point of view 2 highest sounds easiest to maintain.
When we add a new ghc (e.g. 9.10)
- 2 highest will require dropping the older one and adding a new one (e.g. drop 9.6, add 9.10)
- newest+oldest will require updating the oldest to one higher (9.2->9.4) + newest to one higher (9.8->9.10)

So I would vote for "2 highest", but no strong feeling about it.

File: ghcide-bench/src/Experiments.hs [hunk 1]
```diff
@@ -613,21 +623,22 @@ setup = do
...
-            unless alreadyExists $
-                  cmd_ (Cwd exampleDir) (FileStdout hieYamlPath) ("gen-hie"::String)
+            unless alreadyExists $ writeFile hieYamlPath simpleCabalCradleContent
+
           return exampleDir
       ExampleHackage ExamplePackage{..} -> do
         let path = examplesPath </> package
             package = packageName <> "-" <> showVersion packageVersion
             hieYamlPath = path </> "hie.yaml"
+        print hieYamlPath
```
Comment (fendor) [inline ghcide-bench/src/Experiments.hs:633]:
```suggestion
```

debug output?

**Review by fendor — APPROVED** [review]
LGTM, thanks for fixing up the benchmark!

**Comment by michaelpj** [comment]
I think just do two. 51 minutes is a lot, and the benchmark job does actually run quite a bit, so we're just burning CPU cycles for not much benefit.

File: haskell-language-server.cabal [hunk 2]
```diff
@@ -1835,8 +1835,6 @@ test-suite wrapper-test
 benchmark benchmark
     import:           defaults, warnings
     -- Depends on shake-bench which is unbuildable after this point
-    if impl(ghc >= 9.5)
```
[the remark this answers is not in the document]
Comment (soulomoon) [reply haskell-language-server.cabal:1838]: true

File: shake-bench/shake-bench.cabal [hunk 3]
```diff
@@ -17,8 +17,6 @@ source-repository head
 
 library
   -- Depends on Chart which is unbuildable after this point
-  if impl(ghc >= 9.5)
```
[the remark this answers is not in the document]
Comment (soulomoon) [reply shake-bench/shake-bench.cabal:20]: Not true, we can build them, I forget to delete them.

**Comment by soulomoon** [comment]
After hearing you guys opinion.
I decided we keep just two latest versions.
Benchmark is expensive. @jhrcek "maintenance" (lacking man power) and @michaelpj "support longest" (make our development more forward looking) does defeat the need to capture the most different.

**Review by jhrcek — APPROVED** [review]
Thanks! LGTM

**Pull request merged**